### ISSUE TITLE
pyinstaller result is flagged as a threat

### ISSUE THREAD

**Issue by caver456** [ncssar/gpsio#24]
Making a new issue here, to remove detailed discussion of the problem from #5

gpsio-host.exe created from pyinstaller is flagged by windows defender as containing a threat: Trojan:Win32/Sabsik.TE.A!ml - googling 'pyinstaller trojan' shows several folks have reported false positives on pyinstaller results. This one looks like a good place to start:
https://python.plainenglish.io/pyinstaller-exe-false-positive-trojan-virus-resolved-b33842bd3184

**Comment by caver456** [comment]
Following the guidelines at that post.  Screenshot of results from virustotal.com:

![image](https://user-images.githubusercontent.com/18752102/140754440-6391a059-1a95-446c-85be-2552769751de.png)

**Comment by caver456** [comment]
Scan after step 2 (delete build,dist,__pycache__ and run 'pyinstaller gpsio-host.py' again) seems to be hung but does show threats:

![image](https://user-images.githubusercontent.com/18752102/140756430-bbd2566e-f90f-4687-bab9-b29c8acc68a9.png)

**Comment by caver456** [comment]
Looks like it's hung at 899.99sec, but reloading the same file fresh immediately went to the 11/67 summary, so it probably finished the check on their server but the web page didn't update for whatever reason.  Anyway - same results as first pass, with one more test and one more threat:

![image](https://user-images.githubusercontent.com/18752102/140770787-65670cb0-2ccd-4f1c-a9ef-d228ec2350c2.png)

**Comment by caver456** [comment]
Following a suggestion here https://stackoverflow.com/questions/43777106/program-made-with-pyinstaller-now-seen-as-a-trojan-horse-by-avg, pip install pyinstaller==4.1.0 results in a lower count at virusone, but still 8 too many:

![image](https://user-images.githubusercontent.com/18752102/140832195-700b00a7-ac15-4b31-b15e-92208c8a1a9b.png)

**Comment by caver456** [comment]
using py2exe instead gave 15 hits.  And, that was before actually trying the .exe in the workflow - though it did run enough to give a log file output.

A couple other suggestions from that previous link sound intriguing:
- don't include any icons
- make sure everything is on the same 32-vs-64 architecture.  Noticed in the manifest from pyinstaller that it said something about 32.

Tried nuitka.  Only 4 flags, also did not yet confirm that it works in the flow.

**Comment by caver456** [comment]
Seems like the only way to avoid antivirus flags is to not ship an .exe, but to ship the .py instead and require a full python installation.  Seems a bit much...

**Comment by caver456** [comment]
finished all the steps at that link - uninstall, compile from source, rerun.  Still flagged as a threat by 12 vendors:

![image](https://user-images.githubusercontent.com/18752102/140830668-1203069f-bd4d-4bdc-9ee2-2c3ee4861dd3.png)

**Comment by caver456** [comment]
This is a real dilemma.  Maybe doing a 'real' python installation, and then shipping the gpsio-host.py, is the way to go.  Look at [the official python docs](https://docs.python.org/3/using/windows.html), especially the sections on installing without UI (4.1.3), installing without downloading (4.1.4), and embedding (4.4.1)

**Comment by caver456** [comment]
Opened a github question about this dilemma: https://stackoverflow.com/questions/69899867

**Comment by caver456** [comment]
Using the embedded package seems to work so far.  See section 4.4.1 as mentioned above, the 'simpler' option.  Basically, download the embeddable zip from python.org:

https://www.python.org/downloads/release/python-3100/ 'Windows embeddable package (32 bit)'

Then extract that zip file, copy it as the 'dist' directory (or do whatever you want to make it the place that python.exe is called from) and call it from the .bat file like so:

call "%~dp0\dist\python.exe" gpsio-host.py

Windows defender doesn't locate any threats after a scan of the gpsio directory; virustotal.com also does not flag any threats for python.exe - since it's the unmodified exe from python.org.  It seems to run as expected... so far so good... yet to do the entire test from the extension and the NSIS installer...

**Comment by caver456** [comment]
Tested with a GPSIO import from caltopo.com - it works as expected (using the local override main.js file).  This seems like a great option: just use the embeddable package.  The dist dir (i.e. the embeddable package) is 13.6MB extracted, 7.1MB zipped (as downloaded from python.org).

**Comment by caver456** [comment]
Closing for now - it seems to be resolved using the embeddable package straight from the source.
